### PR TITLE
No systemd on nonstandard home

### PR DESCRIPTION
**Pull request by heronhaye**
Continued from https://github.com/keybase/client/pull/19897/files

Test cases
* Specified via cmdline, relative, different from real home
* Specified via cmdline, abs, different from real home
* Specified via envvar, relative, different from real home
* Specified via envvar, abs, different from real home
* Specified via cmdline, abs, same as real home (should connect to systemd if present)
  * This is a behavior change. Is it ok?
* Specified via envvar, abs, same as real home (should connect to systemd if present)

### CODE REVIEW

File: go/libkb/home.go [hunk 1]
```diff
@@ -85,53 +92,63 @@ func (x XdgPosix) Home(emptyOk bool) string {
 	if len(ret) == 0 && !emptyOk {
 		ret = x.getenv("HOME")
 	}
-	return ret
+	if ret == "" {
+		return ""
+	}
+	resolved, err := filepath.Abs(ret)
```
Comment (heronhaye) [inline go/libkb/home.go]: Resolve partial paths (so `HOME=ok keybase id max` works, e.g.)

**Comment by heronhaye** [comment]
cc @maxtaco.

File: go/libkb/home.go [hunk 2]
```diff
@@ -43,6 +45,7 @@ type HomeFinder interface {
 	ServiceSpawnDir() (string, error)
 	SandboxCacheDir() string // For macOS
 	InfoDir() string
+	IsNonstandardHome() (bool, error)
 }
 
 func (b Base) getHome() string {
```
Comment (zapu) [inline go/libkb/home.go]: Unrelated to this PR but having both `Home` and `getHome` in this struct hierarchy is weird.

Did I get it correctly, that `getHome` is solely for getting "configured home" via either config or command line?

Comment (heronhaye) [reply go/libkb/home.go]: Sounds about right

**Pull request merged**